openstack/client.go

```diff
@@ -317,7 +317,10 @@ func NewDNSV2(client *gophercloud.ProviderClient, eo gophercloud.EndpointOpts) (
 	if err != nil {
 		return nil, err
 	}
-	return &gophercloud.ServiceClient{ProviderClient: client, Endpoint: url}, nil
+	return &gophercloud.ServiceClient{
+		ProviderClient: client,
+		Endpoint:       url,
+		ResourceBase:   url + "v2/"}, nil
```

Comment:
@jrperritt is this the correct thing to do? I noticed this was done for some services, but not all. I haven't dug into why (been a little pressed for time) but will do later just so I see why.